onmt/translate/greedy_search.py

```diff
@@ -78,38 +123,44 @@ class GreedySearch(DecodeStrategy):
             :func:`~onmt.translate.greedy_search.sample_with_temperature()`.
         keep_topk (int): See
             :func:`~onmt.translate.greedy_search.sample_with_temperature()`.
+        keep_topp (float): See
+            :func:`~onmt.translate.greedy_search.sample_with_temperature()`.
+        beam_size (int): Number of beams to use.
     """
 
-    def __init__(self, pad, bos, eos, batch_size, min_length,
-                 block_ngram_repeat, exclusion_tokens, return_attention,
-                 max_length, sampling_temp, keep_topk):
+    def __init__(self, pad, bos, eos, unk, batch_size, global_scorer,
+                 min_length, block_ngram_repeat, exclusion_tokens,
+                 return_attention, max_length, sampling_temp, keep_topk,
+                 keep_topp, beam_size, ban_unk_token, always_sample_eos):
         assert block_ngram_repeat == 0
```

Comment:
@Zenglinxiao do you recall why this assertion exists ? It seems like blocking ngrams is supported by GreedySearch

Reply:
At the time of that PR, `block_ngram_repeat` has not properly implemented yet. I think we can remove this assertion now.